tests/lib/predef_heap/src/main.c

```diff
@@ -0,0 +1,565 @@
+/*
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <errno.h>
+#include <zephyr/ztest.h>
+#include <zephyr/sys/predef_heap.h>
+
+ZTEST_SUITE(predef_heap_tests, NULL, NULL, NULL, NULL, NULL);
+
+#define HEAP_MEMORY_BASE	UINT_TO_POINTER(0x10000000)
+
+/* Config store size is 8 * 20 + (8 + 20) * 4 = 272 bytes */
+PREDEFINED_HEAP_DEFINE(test_heap, 8, 513 - BITS_PER_LONG + 4 * BITS_PER_LONG);
+
+static const struct predef_heap_config heap_config[] = {
+	{ .count = 513, .size = 128 },	/* bitfiled size: 17 longs */
+	{ .count = 32, .size = 384 },	/* bitfiled size: 1 long */
+	{ .count = 11, .size = 768 },	/* bitfiled size: 1 long */
+	{ .count = 64, .size = 1536 },	/* bitfiled size: 2 longs */
+	{ .count = 38, .size = 2304 },	/* bitfiled size: 2 longs */
+	{ .count = 60, .size = 3072 },	/* bitfiled size: 2 longs */
+	{ .count = 42, .size = 4224 },	/* bitfiled size: 2 longs */
+	{ .count = 1, .size = 98304 }	/* bitfiled size: 1 long */
+};
+
+/**
+ * @brief Calculate required memory size
+ */
+static size_t calc_mem_size(const struct predef_heap_config *config, int count)
+{
+	size_t size = 0;
+	int i;
+
+	for (i = 0; i < count; i++)
```

Comment:
always keeps me wondering... why not `for(int i = 0; i < count; i++)` ?
Same thing in other source files. In functions with large bodies declarations like those just bloat the stack because you occupy more memory than you need in given moment.